### PR TITLE
feat: support importing blocks without merkle calculation

### PR DESCRIPTION
**Pull request by j75689**
### Description

support importing blocks without merkle calculation mode

### Rationale

Add a flag `--optimize.skip-state-root-validation` to disable state root calculation during block importing. This can be useful if the state root calculation is taking too long.
However, be cautious when using this mode, as the accuracy of node data (PlainState) will no longer be strongly guaranteed.

### Example

1. start a node with `--optimize.skip-state-root-validation` flag
```bash
bsc-reth node --chain bsc --http --http.port 8545 --optimize.skip-state-root-validation
```
(*Optional*) 2. Clear Trie/Hashed Account/Storage table to save database cost.
```
bsc-reth db clear mdbx StoragesTrie
bsc-reth db clear mdbx AccountsTrie
bsc-reth db clear mdbx HashedAccounts
bsc-reth db clear mdbx HashedStorages

bsc-reth db stats
```

### Changes

Notable changes: 
* add new flag to start nodes bypassing merkle root calculation

### Potential Impacts
* no

### CODE REVIEW

File: bin/reth/src/commands/debug_cmd/execution.rs [hunk 1]
```diff
@@ -51,6 +51,9 @@ pub struct Command {
     /// Defaults to `1000`.
     #[arg(long, default_value = "1000")]
     pub interval: u64,
+
+    #[arg(long)]
+    disable_hashing_stages: bool,
```
Comment (unclezoro) [inline bin/reth/src/commands/debug_cmd/execution.rs]: to my understanding, `Merkle stage` will be disabled as well during stage sync?

guess if we can have a better name for this

File: crates/blockchain-tree/src/blockchain_tree.rs [hunk 2]
```diff
@@ -72,6 +72,8 @@ pub struct BlockchainTree<DB, E> {
     sync_metrics_tx: Option<MetricEventsSender>,
     /// Metrics for the blockchain tree.
     metrics: TreeMetrics,
+    /// Disable merkle root calculation for blocks.
+    disable_merkle_root_calculation: bool,
```
Comment (unclezoro) [inline crates/blockchain-tree/src/blockchain_tree.rs]: can we reuse the following name:
```
pub enum BlockValidationKind {
    /// All validation checks that can be performed.
    ///
    /// This includes validating the state root, if possible.
    ///
    /// Note: This should always be used when inserting blocks that originate from the consensus
    /// layer.
    #[default]
    Exhaustive,
    /// Perform all validation checks except for state root validation.
    SkipStateRootValidation,
}
```

File: bin/reth/src/commands/debug_cmd/execution.rs (hunk 1 above)
Comment (j75689) [reply bin/reth/src/commands/debug_cmd/execution.rs]: yes, hashing stages means `HashedAccount/Storage Stage` and `Merkle Stage`

Comment (j75689) [reply bin/reth/src/commands/debug_cmd/execution.rs]: rename to `skip_state_root_validation`

File: crates/blockchain-tree/src/blockchain_tree.rs (hunk 2 above)
Comment (j75689) [reply crates/blockchain-tree/src/blockchain_tree.rs]: rename to `skip_state_root_validation`

File: crates/node/builder/src/setup.rs [hunk 3]
```diff
@@ -36,6 +36,7 @@ pub fn build_networked_pipeline<DB, Client, Executor>(
     static_file_producer: StaticFileProducer<DB>,
     executor: Executor,
     exex_manager_handle: ExExManagerHandle,
+    disable_hash_stages: bool,
```
Comment (unclezoro) [inline crates/node/builder/src/setup.rs]: change this too?

Comment (j75689) [reply crates/node/builder/src/setup.rs]: fixed

**Review by unclezoro — APPROVED** [review]

**Review by pythonberg1997 — APPROVED** [review]

**Pull request merged**